### PR TITLE
fail properly while parsing duplicate tags

### PR DESCRIPTION
**Pull request by harshavardhana**
AWS S3 API expects duplicate keys to be rejected,
due to our usage of url.ParseQuery() this was skipped 
since URL query parser simply appends duplicate keys 
as key with multiple values, this is not desirable.

Add more input validation and restrictions as per
AWS S3 API.

### CODE REVIEW

File: pkg/tags/tags.go [hunk 1]
```diff
@@ -311,25 +341,47 @@ func ParseObjectXML(reader io.Reader) (*Tags, error) {
 	return unmarshalXML(reader, true)
 }
 
+func (tags *tagSet) parseTags(tgs string) (err error) {
+	for tgs != "" {
+		var key string
+		key, tgs, _ = strings.Cut(tgs, "&")
```
Comment (klauspost) [inline pkg/tags/tags.go:347]: Yeah, a bit too soon. When 1.20 is out we can use it :)

Comment (harshavardhana) [reply pkg/tags/tags.go]: done

File: .github/workflows/go-windows.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        go-version: [1.17.x]
+        go-version: [1.18.x, 1.19.x]
```
Comment (klauspost) [inline .github/workflows/go-windows.yml:17]:
```suggestion
        go-version: [1.17.x, 1.18.x, 1.19.x]
```
IMO we should keep 3 active versions for `minio-go`. For other libs (madmin, mc, minio) we can be more aggressive.

Implementing a `strings.Cut` isn't really a big deal.

Comment (harshavardhana) [reply .github/workflows/go-windows.yml:17]: I thought go1.17 is deprecated? @klauspost

Comment (harshavardhana) [reply .github/workflows/go-windows.yml]: Okay got back go1.17 for now.

**Review by klauspost — APPROVED** [review]

**Pull request merged**